server/src/data_store.cpp

```diff
@@ -198,4 +218,37 @@ int DataStore::DbCallback(void* not_used, int argc, char** argv, char** azColNam
     return 0;
 }
 
+std::vector<std::string> DataStore::ReadDistinctMacAddrs(const std::string& device_id)
+{
+    console_->debug("+ DataStore::GetRSSIDataStream");
+
+    std::vector<std::string> mac_addrs;
+    std::string              sql = "SELECT DISTINCT mac_addr FROM  dev_" + device_id + ";";
+
+    sqlite3_stmt* selectStmt;
+    sqlite3_prepare(database_, sql.c_str(), static_cast<int>(sql.length() + 1), &selectStmt, NULL);
+    while (1)
+    {
+        int state = sqlite3_step(selectStmt);
+        if (state == SQLITE_ROW)
+        {
+            std::string mac_addr = reinterpret_cast<const char*>(sqlite3_column_text(selectStmt, 0));
+            mac_addrs.push_back(mac_addr);
+        }
+        else if (state == SQLITE_DONE)
+        {
+            break;
+        }
+        else
+        {
+            console_->error("Failed to read from database");
+            break;
+        }
+    }
+    sqlite3_finalize(selectStmt);
+
+    console_->debug("- DataStore::GetRSSIDataStream");
```

Comment:
In the case of an error, you will get both printouts from line 244 **and** line 250. For example:

`console_->error("Failed to read from database");`
And then before returning:
`console_->debug("- DataStore::GetRSSIDataStream");`

Reply:
Aha, that is intended. The debug print is deactivated should be deactivated when not in debug mode, so we will end up with only the error print. For the debug print of function name however, I prefer to have it at the entry and exit of the function as it is now.

Reply:
Aha, if you want it to always be printed alrighty then!